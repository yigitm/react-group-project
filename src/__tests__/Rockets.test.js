import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';
import Image from 'react-bootstrap/Image';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import 'bootstrap/dist/css/bootstrap.min.css';

describe('Rockets.js: component test', () => {
  test('Rockets.js: should render cancel & book rocket buttons according to reserved value', () => {
    const falcon1 = {
      id: '1',
      rocketName: 'Falcon 1',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: true,
    };
    const falcon9 = {
      id: '2',
      rocketName: 'Falcon 9',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: false,
    };
    const rockets = [falcon1, falcon9];

    const displayRockets = rockets.map((rocket) => (
      <Row key={rocket.id} className="g-1 mb-3">
        <Col className="col-2">
          <Image src={rocket.flickrImages} className="img-thumbnail border-0" />
        </Col>
        <Col className="col-10">
          <Card.Body>
            <Card.Title className="fw-bold">{rocket.rocketName}</Card.Title>
            <Card.Text>
              {rocket.reserved ? (
                <Badge bg="info" className="me-2">
                  Reserved
                </Badge>
              ) : null}
              {rocket.description}
            </Card.Text>
            <Button
              id={rocket.id}
              variant={rocket.reserved ? 'outline-secondary' : 'primary'}
              onClick={(e) => {
                handleReserve(e, rocket);
              }}
            >
              {rocket.reserved ? 'Cancel' : 'Reserve'}
            </Button>
          </Card.Body>
        </Col>
      </Row>
    ));

    const tree = render(
      <Provider store={store}>
        <Container fluid className="mt-5">
          {displayRockets}
        </Container>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('Rockets.js: should return rocket if `reserved: true` to display badge & reserve button', () => {
    const falcon1 = {
      id: '1',
      rocketName: 'Falcon 1',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: true,
    };
    const falcon9 = {
      id: '2',
      rocketName: 'Falcon 9',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: false,
    };

    const rockets = [falcon1, falcon9];
    const badge = [];

    rockets.map((rocket) => {
      rocket.reserved ? badge.push(rocket) : null;
    }),
      expect(badge.length).toEqual(1);
  });

  test('Rockets.js: should return null if `reserved: false` to display cancel button', () => {
    const falcon1 = {
      id: '1',
      rocketName: 'Falcon 1',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: false,
    };
    const falcon9 = {
      id: '2',
      rocketName: 'Falcon 9',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: false,
    };

    const rockets = [falcon1, falcon9];
    const cancelButtons = [];

    rockets.map((rocket) => {
      rocket.reserved ? rocket : cancelButtons.push(null);
    });

    expect(cancelButtons.length).toEqual(2);
  });
});
