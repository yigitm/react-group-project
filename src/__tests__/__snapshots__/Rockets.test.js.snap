// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets.js: component test Rockets.js: should render cancel & book rocket buttons WITHOUT reserved value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5 container-fluid"
      >
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 1
              </div>
              <p
                class="card-text"
              >
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-primary"
                id="1"
                type="button"
              >
                Reserve
              </button>
            </div>
          </div>
        </div>
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 9
              </div>
              <p
                class="card-text"
              >
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-primary"
                id="2"
                type="button"
              >
                Reserve
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5 container-fluid"
    >
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 1
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-primary"
              id="1"
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 9
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-primary"
              id="2"
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rockets.js: component test Rockets.js: should render cancel & book rocket buttons according to reserved value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5 container-fluid"
      >
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 1
              </div>
              <p
                class="card-text"
              >
                <span
                  class="me-2 badge bg-info"
                >
                  Reserved
                </span>
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-outline-secondary"
                id="1"
                type="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </div>
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 9
              </div>
              <p
                class="card-text"
              >
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-primary"
                id="2"
                type="button"
              >
                Reserve
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5 container-fluid"
    >
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 1
            </div>
            <p
              class="card-text"
            >
              <span
                class="me-2 badge bg-info"
              >
                Reserved
              </span>
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-outline-secondary"
              id="1"
              type="button"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 9
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-primary"
              id="2"
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rockets.js: component test Rockets.js: should render cancel & book rocket buttons according to reserved value 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5 container-fluid"
      >
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 1
              </div>
              <p
                class="card-text"
              >
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-primary"
                id="1"
                type="button"
              >
                Reserve
              </button>
            </div>
          </div>
        </div>
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 9
              </div>
              <p
                class="card-text"
              >
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-primary"
                id="2"
                type="button"
              >
                Reserve
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5 container-fluid"
    >
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 1
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-primary"
              id="1"
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 9
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-primary"
              id="2"
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rockets.js: component test Rockets.js: should render cancel & book rocket buttons without any reserved value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5 container-fluid"
      >
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 1
              </div>
              <p
                class="card-text"
              >
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-primary"
                id="1"
                type="button"
              >
                Reserve
              </button>
            </div>
          </div>
        </div>
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 9
              </div>
              <p
                class="card-text"
              >
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-primary"
                id="2"
                type="button"
              >
                Reserve
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5 container-fluid"
    >
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 1
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-primary"
              id="1"
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 9
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-primary"
              id="2"
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rockets.js: page test Rockets.js: should load  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5 container-fluid"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5 container-fluid"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rockets.js: page test Rockets.js: should load  2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5 container-fluid"
      >
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 1
              </div>
              <p
                class="card-text"
              >
                <span
                  class="me-2 badge bg-info"
                >
                  Reserved
                </span>
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-outline-secondary"
                id="1"
                type="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </div>
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 9
              </div>
              <p
                class="card-text"
              >
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-primary"
                id="2"
                type="button"
              >
                Reserve
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5 container-fluid"
    >
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 1
            </div>
            <p
              class="card-text"
            >
              <span
                class="me-2 badge bg-info"
              >
                Reserved
              </span>
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-outline-secondary"
              id="1"
              type="button"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 9
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-primary"
              id="2"
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rockets.js: page test Rockets.js: should load cancel & book rocket buttons according to reserved value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5 container-fluid"
      >
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 1
              </div>
              <p
                class="card-text"
              >
                <span
                  class="me-2 badge bg-info"
                >
                  Reserved
                </span>
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-outline-secondary"
                id="1"
                type="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </div>
        <div
          class="g-1 mb-3 row"
        >
          <div
            class="col-2 col"
          >
            <img
              class="img-thumbnail border-0"
              src="image_url"
            />
          </div>
          <div
            class="col-10 col"
          >
            <div
              class="card-body"
            >
              <div
                class="fw-bold card-title h5"
              >
                Falcon 9
              </div>
              <p
                class="card-text"
              >
                Lorem ipsum onum
              </p>
              <button
                class="btn btn-primary"
                id="2"
                type="button"
              >
                Reserve
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5 container-fluid"
    >
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 1
            </div>
            <p
              class="card-text"
            >
              <span
                class="me-2 badge bg-info"
              >
                Reserved
              </span>
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-outline-secondary"
              id="1"
              type="button"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
      <div
        class="g-1 mb-3 row"
      >
        <div
          class="col-2 col"
        >
          <img
            class="img-thumbnail border-0"
            src="image_url"
          />
        </div>
        <div
          class="col-10 col"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold card-title h5"
            >
              Falcon 9
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum onum
            </p>
            <button
              class="btn btn-primary"
              id="2"
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
